### PR TITLE
Refactor universe objects to pass objects as the parameter

### PR DESCRIPTION
**Pull request by quittle**
Refactor universe objects to pass objects as the parameter

https://user-images.githubusercontent.com/671716/132107838-11264db6-9be1-45c0-95c9-64d550878e34.mp4

### CODE REVIEW

File: src/ts/universe.ts [hunk 1]
```diff
@@ -47,16 +55,29 @@ export class UniverseObject {
...
 /** An object with mass, but does not require that it takes up space. E.g. black holes */
 export class UniverseObjectWithMass extends UniverseObject {
   mass: number;
 
-  constructor(x: number, y: number) {
-    super(x, y);
-    this.mass = 0;
+  constructor(options: UniverseObjectWithMassOptions) {
+    super(options);
+    this.mass = options.mass ?? 0;
```
Comment (bschlenk) [inline src/ts/universe.ts:68]: I'd just use `||` here, `??` transpiles down to `options.mass !== null && options.mass !== undefined ? options.mass : 0`

File: src/ts/universe.ts [hunk 2]
```diff
@@ -66,12 +87,12 @@ export class UniverseCircle extends UniverseObjectWithMass {
...
-    this.radius = 10;
-    this.texture = '#fff';
-    this.orientation = 0;
-    this.rotationSpeed = 0;
+  constructor(options: UniverseCircleOptions) {
+    super(options);
+    this.radius = options.radius ?? 10;
+    this.texture = options.texture ?? '#fff';
+    this.orientation = options.orientation ?? 0;
+    this.rotationSpeed = options.rotationSpeed ?? 0;
```
Comment (bschlenk) [inline src/ts/universe.ts:95]: For the same reason as above I wouldn't use `??`. You could do

```
constructor({ radius = 10, texture = '#fff', orientation = 0, rotationSpeed = 0, ...options }: UniverseCircleOptions) {
  super(options);
  this.radius = radius;
  ...
}
```

File: src/ts/vector.ts [hunk 3]
```diff
@@ -17,3 +17,7 @@ export function vecAngleBetween(a: Vector, b: Vector) {
 export function vecFromAngleAndScale(radians: number, scale = 1) {
   return { x: Math.cos(radians) * scale, y: Math.sin(radians) * scale };
 }
+
+export function vecEquals(a: Vector, b: Vector): boolean {
+  return a.x === b.x && a.y === b.y && a.dx === a.dx && a.dy === b.dy;
```
Comment (bschlenk) [inline src/ts/vector.ts:22]: We wouldn't have one vector with all 4 of these properties would we? I think we should change velocity to use x and y, there's no reason to use different properties

**Review by bschlenk — APPROVED** [review]

File: src/ts/universe.ts (hunk 1 above)
Comment (quittle) [reply src/ts/universe.ts:68]: The nice thing is, we aren't transpiling so we get the literal `??` in the output, so it doesn't take up more space.

File: src/ts/vector.ts (hunk 3 above)
Comment (quittle) [reply src/ts/vector.ts:22]: I also am not sure why we have a vector with `dx` and `dy` but given that we do, this followed the principle of least surprise to check all of them. I'm fully in favor of splitting velocity away from position and having just one vector.

**Comment by quittle** [comment]
Merged as part of #36

**Pull request closed without merging**